Perhaps consider going the ES6 route.

SaulDoesCode opened the pull request:
I converted this awesome animation library to es6 code added and custom build system it works exactly the same.
##### benefits 
* es6 code
* less lines of code
* build system - minifies and transpiles to es5

maybe look over make some suggestions.
say if it's bad or if you like the concept but perhaps change it your way.

Maximilianos commented on build.js:
```diff
@@ -0,0 +1,113 @@
+    });
+}
+
+function BabelizeBeatifyMinify(inloc, outloc) {
```
should be BabelBea**u**tifyMinify :)

SaulDoesCode commented:
There still seems to be one or two bugs I'm working out with the conversion. I'm testing it on your Code Pen demos. thanks for looking at it will apply your suggestion for the build.js file. will keep you informed on whats happening. still working out some kinks

Maximilianos commented on package.json:
```diff
@@ -1,11 +1,11 @@
   "repository": {
     "type": "git",
-    "url": "https://github.com/juliangarnier/anime.git"
+    "url": "https://github.com/sauldoescode/anime.git"
```
prob want to keep julian garnier here :) and same below

SaulDoesCode replied:
yeah sorry lol not used to working with git.

Maximilianos replied:
no worries :)

SaulDoesCode commented:
fixed silly bug with toArray, all the demos are working in ES6 version so far.

tbranyen commented on .gitignore:
```diff
@@ -0,0 +1,6 @@
+node_modules
+
+#FileTypes
+*.exe
```
Why do you need to ignore Windows executables in this library?

SaulDoesCode replied:
Just copied it from one of my projects that uses .exe files. I'll remove it

SaulDoesCode commented:
Perhaps users may benefit from exposing some internal utility functions such as includes, removeArrayDuplicates, flattenArray like they do with anime.random.

lukeed commented:
I'd prefer **not** to see this happen 👎 

There's nothing wrong with the current distribution, and adding in ES6 only adds build complexity & usually transpilation "bloat". Your branch doesn't split the code into files, thus `require`, `import`, and `export` aren't used, but once a codebase is written in ES6, that's generally the next step.

---

Sidenote: @SaulDoesCode future note, your `package.json` should be declaring `devDependencies`, not `dependencies`.

SaulDoesCode commented:
Thanks, I understand. I'll try to keep this branch updated and in sync with the main repo for whom ever it matters. Will change that in the package.json thanks.

lukeed commented on anime.js:
```diff
@@ -1,605 +1,555 @@
-      setWillChange(anim);
-      animations.push(anim);
-      time.raf = requestAnimationFrame(time.tick);
+    typeof define === 'function' && define.amd ? define([], factory) :
```
incorrect ternary declarations. all these will run

SaulDoesCode replied:
thanks for noticing will change that.

lukeed replied:
It'd be a multi-nested ternary expression, which is hard to follow. And you can't use a `return` for relief either.. so just copy the original `if/elif/else` block; it's common across most javascript packages.

lukeed commented on anime.js:
```diff
@@ -1,605 +1,555 @@
+    // Defaults
+    const undef = void 0;
+
+    let defaultSettings = {
```
use const, doesnt change

lukeed commented on anime.js:
```diff
@@ -1,605 +1,555 @@
-            if (!animation.tweens.length) animation.pause();
-          }
+    const is = (() => ({
+        array: Array.isArray,
```
should be a function of `a`

SaulDoesCode replied:
it is wrapped in a function in the original repo but it doesn't change the behavior this way Array.isArray should be fine.

lukeed commented on anime.js:
```diff
@@ -1,605 +1,555 @@
+
+    function includes(arr, searchElement) {
+        if(arr.includes) return arr.includes(searchElement);
+        if (!Array.isArray(arr)) arr = Array.prototype.slice.call(arr);
```
might as well use the `is` util: 

```js
if (!is.array(arr)) //..
```

lukeed commented on anime.js:
```diff
@@ -1,605 +1,555 @@
+    function includes(arr, searchElement) {
+        if(arr.includes) return arr.includes(searchElement);
+        if (!Array.isArray(arr)) arr = Array.prototype.slice.call(arr);
+        return arr.length == 0 ? false : arr.some(item => item === searchitem);
```
triple `===`

lukeed commented on anime.js:
```diff
@@ -1,605 +1,555 @@
+    // Easings functions adapted from http://jqueryui.com/
+
+    const easings = (() => {
+        let eases = {},
```
split these variables, some are constants

SaulDoesCode commented:
Awesome constructive criticism @lukeed thanks I'll see to it that those changes be implemented

lukeed commented on anime.js:
```diff
@@ -1,605 +1,555 @@
+
+    // Strings
+
+    const numberToString = val => (is.string(val)) ? val : val + '',
```
template string: wrap `${val}` in "`"

lukeed commented on anime.js:
```diff
@@ -1,605 +1,555 @@
+            return [o];
+        },
+
+        flattenArray = arr => Array.prototype.reduce.call(arr, (a, b) => a.concat(is.array(b) ? flattenArray(b) : b), []),
```
leave as `arr.reduce`

SaulDoesCode replied:
flattenArray doesn't always receive pure arrays all the time so it makes errors this way it works with arraylike objects

lukeed commented on anime.js:
```diff
@@ -1,605 +1,555 @@
+        groupArrayByProps = (arr, propsArr) => {
+            let groups = {};
+            arr.forEach(o => {
+                let group = JSON.stringify(propsArr.map(p => o[p]));
```
constant

lukeed commented on anime.js:
```diff
@@ -1,605 +1,555 @@
+        // Colors
+        hexToRgb = hex => {
+            hex = hex.replace(/^#?([a-f\d])([a-f\d])([a-f\d])$/i, (m, r, g, b) => r + r + g + g + b + b);
+            let rgb = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex),
```
constants

lukeed commented on anime.js:
```diff
@@ -1,605 +1,555 @@
+                r = parseInt(rgb[1], 16),
+                g = parseInt(rgb[2], 16),
+                b = parseInt(rgb[3], 16);
+            return 'rgb(' + r + ',' + g + ',' + b + ')';
```
es6 template string

lukeed commented:
@SaulDoesCode There's more that I can add, but you'd receive the same critiques from [XO](https://github.com/sindresorhus/xo). You might want to check that out instead :)

SaulDoesCode commented:
Very well I'll see how that linter works and follow it's guide lines. again many thanks for helping

lukeed commented:
@SaulDoesCode No problem 👌 Ping me if you need help setting up XO. It's a great tool

SaulDoesCode commented:
Keeping up with all the changes so far, implemented latest from anime v1.1.0 and implemented @devmario's chainable suggestions.
nice work with the engine object and usage of performance.now so much better now. :+1:

devmario commented on anime.js:
```diff
@@ -534,14 +535,27 @@
+                      anim[type + '_resolve'] = resolve;
+                    });
+                }
+                console.warn('Your browser doesn\'t support promises.');
```
This warning msg sounds good more then before!
